civicrm/custom/php/CRM/NYSS/BAO/Integration/WebsiteEvent.php

```diff
@@ -248,16 +249,18 @@ protected function createEntityTag(int $contact_id, int $tag_id): void {
    * @throws \Civi\API\Exception\UnauthorizedException
    */
   protected function deleteEntityTag(int $contact_id, int $tag_id): void {
-    $results = \Civi\Api4\EntityTag::delete($this->getCiviPermissionCheck())
-      ->addWhere('entity_table', '=', 'civicrm_contact')
-      ->addWhere('entity_id', '=', $contact_id)
-      ->addWhere('tag_id', '=', $tag_id)
-      ->setCheckPermissions($this->getCiviPermissionCheck())
-      ->execute();
-
-    if (count($results) < 1) {
-      throw new \CRM_Core_Exception('Failed to delete entity tag');
-    }
+      if ($this->hasEntityTag($contact_id,$tag_id)) {
+          $results = \Civi\Api4\EntityTag::delete($this->getCiviPermissionCheck())
+              ->addWhere('entity_table', '=', 'civicrm_contact')
+              ->addWhere('entity_id', '=', $contact_id)
+              ->addWhere('tag_id', '=', $tag_id)
+              ->setCheckPermissions($this->getCiviPermissionCheck())
+              ->execute();
+
+          if (count($results) < 1) {
+              throw new \CRM_Core_Exception('Failed to delete entity tag');
```

Comment:
I tend to avoid throwing exceptions unless I really need everything to fully stop. In this case I'd be more inclined to just log an error.

Reply:
@lcdservices Please note that this exception is bubbling to the top (process.php) where process.php is doing what you've suggested -- logging the error and continuing execution.

I'd appreciate if you'd let this one through as is because we're getting pressure from the top to finish this up asap.